netket/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,644 @@
+            >>> op
+            >>> op.hilbert
+            >>> op.hilbert.size
+        """
+        super().__init__(hilbert)
+        self._dtype = dtype
+        self._orb_idxs = []
+        self._daggers = []
+        self._term_ends = []
+        self._weights = []
+        self._n_terms = 0
+        # we keep the input, in order to be able to add terms later
+        self._orig_terms = []
+        self._orig_weights = []
+        self._constant = constant
+
+        if isinstance(terms, str):
+            terms = (terms,)
+
+        if len(terms) > 0 and isinstance(terms[0], str):
+            terms = list(map(_parse_string, terms))
+
+        if weights is None:
+            weights = [1.0] * len(terms)
+
+        if not len(weights) == len(terms):
+            raise ValueError("length of weights should be equal")
+
+        for term, weight in zip(terms, weights):
+            self.add_term(term, weight=weight)
+
+        _check_tree_structure(self._orig_terms)
+
+        self._initialized = False
+        self._is_hermitian = _check_hermitian(self._orig_terms, self._orig_weights)
+
+    def add_term(self, term, weight=1.0):
+        if isinstance(term, str):
+            term = _parse_string(term)
+
+        self._orig_terms.append(term)
+        self._orig_weights.append(weight)
+        if len(term) == 0:
+            raise ValueError("terms cannot be size 0")
+        if not all(len(t) == 2 for t in term):
+            raise ValueError(
+                "terms must contain (i, dag) pairs, but received {}".format(term)
+            )
+        for orb_idx, dagger in reversed(term):
+            self._orb_idxs.append(orb_idx)
+            self._daggers.append(bool(dagger))
+            self._weights.append(weight)
+            self._term_ends.append(False)
+        self._term_ends[-1] = True
+        self._n_terms += 1
+
+    def _setup(self, force=False):
+        """Analyze the operator strings and precompute arrays for get_conn inference"""
+        if force or not self._initialized:
+            self._orb_idxs = np.array(self._orb_idxs, dtype=np.intp)
+            self._daggers = np.array(self._daggers, dtype=bool)
+            self._weights = np.array(self._weights, dtype=self.dtype)
+            self._term_ends = np.array(self._term_ends, dtype=bool)
+            self._initialized = True
+
+    @staticmethod
+    def destroy(
+        hilbert: AbstractHilbert, site: int, sz: int = None, dtype: DType = complex
+    ):
+        """
+        Builds the fermion destruction operator :math:`\\hat{a}` acting on the `site`-th of
+        the Hilbert space `hilbert`.
+        Args:
+            hilbert: The hilbert space
+            site: the site on which this operator acts
+        Returns:
+            The resulting FermionOperator2nd
+        """
+        idx = _get_index(hilbert, site, sz)
+        return FermionOperator2nd(hilbert, ("{}".format(idx),), dtype=dtype)
```

Comment:
this, and functions below: can't we put them into `nk.operator.fermion.create`/`destroy`/ecc like the other operators?